modules/vpc/main.tf

```diff
@@ -20,25 +13,25 @@ resource "aws_vpc" "main" {
 
 #Subnets
 resource "aws_subnet" "main-public" {
-  count = 2
+  count = length(var.public_cidr_block)
 
   vpc_id                  = aws_vpc.main.id
-  cidr_block              = local.public_cidr_block[count.index]
+  cidr_block              = var.public_cidr_block[count.index]
   map_public_ip_on_launch = "true"
-  availability_zone       = local.az[count.index]
+  availability_zone       = var.az[count.index]
 
   tags = {
     Name = "main-public-${count.index + 1}"
```

Comment:
```suggestion
    Name = "${var.env_code}-public-${count.index + 1}"
```

Reply:
Please do the same change the Name tag of each resource

Reply:
This is required for scalability. If the name is not parametrized we will end up with several VPCs called main when we deploy more than one VPC

Reply:
fixed